File: oracles/Java Oracles/01 - Maximum Value/Max.java
public class Max {

	public static int maximum(int[] values) {
		int max = values[0];
		for(int i = 1; i < values.length; i++)
			if (max < values[i])
				max = values[i];
		return max;
	}

	public static int max(int[] values) throws IllegalArgumentException {
		if (values.length == 0)
			throw new IllegalArgumentException();
		else
			return maximum(values);
	}

}

File: progtest-web/resources/oracles/C/Max/Max.c
#include "Max.h"

short int maximum(short int *values, unsigned char n) {
	short int max = values[0];
	unsigned char i;
	for (i = 1; i < n; i++)
		if (max < values[i])
			max = values[i];
	return max;
}

short int *max(short int *values, unsigned char n) {
	short int *max = (short int *) calloc(2, sizeof(short int));
	if (n > 0) {
		max[0] = 1;
		max[1] = maximum(values, n);
	} else {
		max[0] = 0;
	}
	return max;
}
